pyproject.toml

```diff
@@ -36,11 +36,13 @@ sphinx-autobuild = "^2021.3.14"
 matplotlib = "^3.8.2"
 numpydoc = "^1.6.0"
 sphinx-copybutton = "^0.5.2"
-sphinx-exec-code = "^0.10"
+sphinx-exec-code = "^0.12"
 sphinx-rtd-theme = "^1.3.0"
 insegel = "^1.3.1"
 nbsphinx = "^0.9.3"
 pandoc = "^2.3"
+IPython = "8.23.0"
```

Comment:
Can you add IPython using poetry? So `poetry add IPython -G docs`? Then it will not be pinned to specific version like it is here.